packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -767,12 +757,12 @@ export const KeyboardNavigationMixin = (superClass) =>
       e.preventDefault();
 
       const element = e.composedPath()[0];
-      const isRow = this.__isRow(element);
-      if (isRow || !element._content || !element._content.firstElementChild) {
+      const type = isRow(element) ? 'row' : 'cell';
+      if (type === 'row' || !element._content || !element._content.firstElementChild) {
         this.dispatchEvent(
-          new CustomEvent(isRow ? 'row-activate' : 'cell-activate', {
+          new CustomEvent(`${type}-activate`, {
             detail: {
-              model: this.__getRowModel(isRow ? element : element.parentElement),
+              model: this.__getRowModel(type === 'row' ? element : element.parentElement),
```

Comment:
Nit: Not sure if this is a turn for the better. The updated version compares and concatenates string values whereas the current version only has the full event names as string literals (maybe those could be extracted as file-level constants).

Reply:
Good point, I'll revert this change.